### ISSUE TITLE
bug: iOS - ion-back-button in ion-toolbar lacks spacing in end and primary slots

### ISSUE THREAD

**Issue by remisture** [ionic-team/ionic-framework#18176]
# Bug Report

**Ionic version:**
[x] **4.3.1**

**Current behavior:**
In an `ion-toolbar`, when placing an `ion-back-button` in the `end` slot, the button has no spacing applied to it. It is squeezed all the way to the right, and the size is larger than expected. The same applies to the `primary` slot.

**Expected behavior:**
I expect `ion-back-button` to behave the same way as `ion-icon` - see screenshots below.

**Related code:**

`ion-back-button` lacks margin:
```
<ion-header>
	<ion-toolbar color="dark">
		<ion-buttons slot="start">
			<ion-menu-button></ion-menu-button>
		</ion-buttons>
		<ion-title>My header</ion-title>
		<ion-buttons slot="end">
			<ion-back-button [defaultHref]="defaultHref" text="" icon="paper"></ion-back-button>
		</ion-buttons>
	</ion-toolbar>
</ion-header>
```
I expect the `ion-back-button` to behave like an `ion-button`

```
<ion-header>
	<ion-toolbar color="dark">
		<ion-buttons slot="start">
			<ion-menu-button></ion-menu-button>
		</ion-buttons>
		<ion-title>My header</ion-title>
		<ion-buttons slot="end">
			<ion-button>
				<ion-icon name="paper"></ion-icon>
			</ion-button>
		</ion-buttons>
	</ion-toolbar>
</ion-header>
```
**Other information:**
Bug:
<img width="439" alt="bug" src="https://user-images.githubusercontent.com/2372967/57014962-ae3daf00-6c12-11e9-8f64-e26d2b5a911d.png">
Expected:
<img width="437" alt="expected" src="https://user-images.githubusercontent.com/2372967/57014963-ae3daf00-6c12-11e9-8254-757ba8f5996c.png">

**Ionic info:**

```
Ionic:

   ionic (Ionic CLI)             : 4.12.0 (/Users/remisture/.nvm/versions/node/v10.15.3/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.3.1
   @angular-devkit/build-angular : 0.13.8
   @angular-devkit/schematics    : 7.3.8
   @angular/cli                  : 7.3.8
   @ionic/angular-toolkit        : 1.5.1

Cordova:

   cordova (Cordova CLI) : 8.1.2 (cordova-lib@8.1.1)
   Cordova Platforms     : android 7.1.4, ios 4.5.5
   Cordova Plugins       : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 2.4.1, (and 13 other plugins)

System:

   ios-deploy : 1.9.4
   NodeJS     : v10.15.3 (/Users/remisture/.nvm/versions/node/v10.15.3/bin/node)
   npm        : 6.9.0
   OS         : macOS Mojave
   Xcode      : Xcode 10.2.1 Build version 10E1001
```

**Comment by brandyscarney** [comment]
Thanks for the issue! I could reproduce: https://codepen.io/brandyscarney/pen/NmQxZZ?ionic:mode=ios

**Comment by SavageCore** [comment]
I'm seeing the same with any `ion-icon` in slot end? Is that right?

Ionic Info:
```
Ionic:

   Ionic CLI                     : 6.1.0 (/usr/local/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 5.0.0
   @angular-devkit/build-angular : 0.803.1
   @angular-devkit/schematics    : 8.3.1
   @angular/cli                  : 8.3.1
   @ionic/angular-toolkit        : 2.0.0

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : android 8.1.0, ios 5.0.1
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 4.0.1, (and 11 other plugins)

Utility:

   cordova-res : 0.9.0
   native-run  : 0.3.0

System:

   Android SDK Tools : 26.1.1 (/Users/oliversayers/Library/Android/sdk)
   ios-deploy        : 1.10.0
   ios-sim           : 8.0.2
   NodeJS            : v12.13.0 (/usr/local/bin/node)
   npm               : 6.13.7
   OS                : macOS Catalina
   Xcode             : Xcode 11.3.1 Build version 11C504
```

**Comment by thetaPC** [comment]
Closing this issue since it should have been addressed through ionic-team/ionic-framework#27042. Please update to `v7.0.1` for the fix. If you encounter any issues, then please create a new issue.
